lib/OpenQA/WebAPI.pm

```diff
@@ -394,7 +394,10 @@ sub startup {
     $api_public_r->get('job_templates')->name('apiv1_job_templates')->to('job_template#list');
     $api_ra->post('job_templates')->to('job_template#create');
     $api_public_r->get('job_templates/:job_template_id')->name('apiv1_job_template')->to('job_template#list');
-    $api_ra->delete('job_templates/:job_template_id')->to('job_template#destroy');
+    $api_ra->put('job_templates/<job_template_id:num>')->to('job_template#update_properties');
+    $api_ra->post('job_templates/<job_template_id:num>')->to('job_template#update_properties')
+      ;                                                                             #in case PUT is not supported
```

Comment:
There's existing code doing the same thing.. but my question would be, what's the use case? And would adding a slightly obscure extra argument be sufficient for that use case?

Reply:
Browsers only support `GET` and `POST`, so methods like `PUT` always need to be emulated somehow. Using a query parameter like `?_method=PUT` is not obscure at all, it's actually a standard way many frameworks use to make REST APIs available to browsers.

Reply:
Perhaps I should've phrased it differently... we should document if we want to use PUT and/or POST, and if we need both it should be documented. I'm saying "obscure" here really to mean it's not documented, or if it is I haven't seen it.

Reply:
The name `update_properties` actually makes me think this might be a case for `PATCH`. 😆

Reply:
I removed the line - if people can use `?_method=PUT` there is really no need for this. However, I kept the method `PUT` because that's what we already use everywhere else for such cases. Otherwise I would have to add an extra `if` in `load_templates` just for job templates.